### DISCUSSION TITLE
sobrepor um widget em outro

### DISCUSSION THREAD

**Discussion by alanirigotti** [Flutterando/forum#138] [Perguntas e Respostas]
Como colocar uma imagem em cima de um container em flutter? 

Eu criei uma tela de login que possui uma imagem no topo e um container com as bordas arrendondas ocupando mais de metade da tela, entretanto a imagem fica no limite do container e eu gostaria que o container ficasse na frente. Como posso arrumar? 

Segue o código da imagem: 

class LoginPage extends StatelessWidget {
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Stack(
          children: [
            Container(
              height: MediaQuery.of(context).size.height * 0.30,
              decoration: const BoxDecoration(
                image: DecorationImage(
                  image: AssetImage('assets/imagens/idososLogin.png'),
                  fit: BoxFit.cover,
                ),
              ),
            ),

**Comment by edugemini** [comment]
Olá @alanirigotti,
    Apesar de utilizar o widget Stack, você não está utilizando o poder dele de sobreposição, pois está utilizando a imagem como propriedade do Container e não como um elemento separado do array children.
   Como quer que o container fique na frente, o primeiro elemento do array terá que ser a imagem e o segundo o container.
   Exemplo:
```dart
Stack(
  children: [
    AssetImage(...),
    Container(...),
  ]
)
```
Dica 1: Dê uma olhada no widget Positioned para posicionar os elementos em um Stack
Dica 2: Quando for colocar código no Github, na linha anterior ao código utilize três crases e a linguagem do seu código e na linha após seu código coloque novamente três crases (veja no preview como vai ficar antes de postar). Exemplo:

\`\`\`dart
  Seu código aqui
\`\`\`

**Comment by alanirigotti** [comment]
Agora a imagem fica esticada no fundo da tela. Estava conversando com um colega e ele mencionou a herança do dart, ou seja, a imagem se adapta ao fundo. Entretanto ainda não onsegui resolver esse problema. Como deixar sobreposto o container em cima da parte final da imagem  sem que ela fique esticada e no tamanho proporcional?


```    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Stack(
          children: [
            Positioned.fill(
              //heigth: MediaQuery.of(context).size.height * 0.4,
              child: Image.asset('assets/imagens/idososLogin.png',
                  fit: BoxFit.cover),
            ),
            Positioned(
              bottom: 0,
              left: 2,
              right: 0,
              child: Container(
                height: MediaQuery.of(context).size.height * 0.6,
                decoration: const BoxDecoration(
                  borderRadius: BorderRadius.only(
                    topLeft: Radius.circular(30),
                    topRight: Radius.circular(30),
                  ),
                  gradient: LinearGradient(
                    colors: [
                      Colors.white,
                      Color.fromRGBO(48, 103, 184, 1),
                    ],
                    begin: Alignment.bottomCenter,
                    end: Alignment.topCenter,
                  ),
                ),
                child: Padding(
                  padding: const EdgeInsets.only(top: 40),
                  child: Column(
                    children: [
                      const Text(
                        'DIAGNÓSTICO PARA IDOSO',
                        style: TextStyle(
                          fontSize: 20,
                          fontFamily: 'Futura',
                          color: Colors.white,
                        ),
                      ),
                      const SizedBox(height: 20), 
````
